File: language/benchmarks/tokens/algorithms/insertion-sort/insertion-sort.ts
function insertSorted(item: number, xs: number[]): number[] {
  if (xs.length === 0) return [item];
  const [head, ...tail] = xs;
  if (item <= head) return [item, head, ...tail];
  return [head, ...insertSorted(item, tail)];
}

function insertionSort(xs: number[]): number[] {
  if (xs.length === 0) return [];
  const [head, ...tail] = xs;
  return insertSorted(head, insertionSort(tail));
}

File: language/benchmarks/tokens/algorithms/insertion-sort/insertion-sort.py
def insert_sorted(item: int, xs: list[int]) -> list[int]:
    if len(xs) == 0:
        return [item]
    head, *tail = xs
    if item <= head:
        return [item, head] + tail
    return [head] + insert_sorted(item, tail)


def insertion_sort(xs: list[int]) -> list[int]:
    if len(xs) == 0:
        return []
    head, *tail = xs
    return insert_sorted(head, insertion_sort(tail))
